### PR TITLE
Implement io.Writer interface

### LINKED ISSUE #4: Add `io.Writer` method

### PR DESCRIPTION
**Pull request by codegoalie**
Closes #4

### CODE REVIEW

File: rope.go [hunk 1]
```diff
@@ -262,3 +262,14 @@ func (r *Rope) Before(at int, fn func(r rune) bool) (retVal int, retRune rune, e
 	}
 	return
 }
+
+// Write implements the io.Writer interface for a Rope. Existing contents of the Rope will be erased.
+func (r *Rope) Write(p []byte) (int, error) {
+	_ = r.EraseAt(0, r.Size())
```
Comment (chewxy) [inline rope.go]: Is there a reason for erasing it? Typical `io.Writer`s don't erase, just append

**Review by chewxy — CHANGES_REQUESTED** [review]
Thank you so much for this pull request. I'm unsure about the decision to erase, rather than just append. I was rather hoping you could help me understand this one

**Comment by codegoalie** [comment]
@chewxy You are right. Multiple calls to the same `Writer` would just append. I've fixed that in 2100014.

**Review by chewxy — APPROVED** [review]
thank you very much

**Pull request merged**